packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1633,13 +1634,39 @@ class DropdownButtonFormField<T> extends FormField<T> {
              }
            }
            final bool isEmpty = !showSelectedItem && !isHintOrDisabledHintAvailable();
+           final bool hasError = effectiveDecoration.errorText != null;
 
            // An unfocusable Focus widget so that this widget can detect if its
            // descendants have focus or not.
            return Focus(
              canRequestFocus: false,
              skipTraversal: true,
              child: Builder(builder: (BuildContext context) {
+                InputBorder? resolveInputBorder() {
+                  if (hasError) {
+                    if (Focus.of(context).hasFocus) {
+                      return effectiveDecoration.focusedErrorBorder;
+                    }
+                    return effectiveDecoration.errorBorder;
+                  }
+                  if (Focus.of(context).hasFocus) {
+                    return effectiveDecoration.focusedBorder;
+                  }
+                  if (effectiveDecoration.enabled) {
+                    return effectiveDecoration.enabledBorder;
+                  }
+                  return effectiveDecoration.border;
+                }
+                BorderRadius? effectiveBorderRadius() {
+                  if (resolveInputBorder() is OutlineInputBorder) {
```

Comment:
What if you put the resolved input border in a variable here? Then you can avoid calling resolve 4x and you'd also get type promotion on it, which would eliminate the `as` typecasts here.

Reply:
Good idea!